multus_sriov_cni.sh

```diff
@@ -0,0 +1,249 @@
+
+function configure_multus {
+    echo "Configure Multus"
+    interface=$1
+
+    kubectl delete -f $WORKSPACE/multus-cni/images/multus-daemonset.yml
+    kubectl create -f $WORKSPACE/multus-cni/images/multus-daemonset.yml
+    sleep 60
+    cat > $CNI_CONF_DIR/00-multus.conf <<EOF
+{
+    "name": "multus-cni-network",
+    "type": "multus",
+    "capabilities": {
+        "portMappings": true
+    },
+    "delegates": [
+        {
+            "cniVersion": "0.3.0",
+            "name": "macvlan-network",
+            "type": "macvlan",
+            "master": "$interface",
+            "mode": "bridge",
+            "ipam": {
+                "type": "host-local",
+                "subnet": "192.168.1.0/24",
+                "rangeStart": "192.168.1.200",
+                "rangeEnd": "192.168.1.216",
+                "routes": [{"dst": "0.0.0.0/0"}],
+                "gateway": "192.168.1.1"
+            }
+        }
+    ],
+    "logFile": "$LOGDIR/multus.log",
+    "logLevel": "debug",
+    "kubeconfig": "$KUBECONFIG"
+}
+EOF
+    cp $CNI_CONF_DIR/00-multus.conf $ARTIFACTS
+    return $?
+}
+
+
+function download_cni {
+
+    if ! $RECLONE ; then
+        return 0
+    fi
+
+    echo "Download $MULTUS_CNI_REPO"
+    git clone $MULTUS_CNI_REPO $WORKSPACE/multus-cni
+    cd $WORKSPACE/multus-cni
+    git checkout $MULTUS_CNI_BRANCH
+    git log -p -1 > $ARTIFACTS/multus-cni.txt
+    cd -
+
+    echo "Download $SRIOV_CNI_REPO"
+    git clone $SRIOV_CNI_REPO $WORKSPACE/sriov-cni
+    pushd $WORKSPACE/sriov-cni
+    git checkout $SRIOV_CNI_BRANCH
+    git log -p -1 > $ARTIFACTS/sriov-cni.txt
+    popd
+
+    echo "Download $PLUGINS_REPO"
+    git clone $PLUGINS_REPO $WORKSPACE/plugins
+    pushd $WORKSPACE/plugins
+    git checkout $PLUGINS_BRANCH
+    git log -p -1 > $ARTIFACTS/plugins.txt
+    ./build_linux.sh
+    popd
+
+    echo "Download $SRIOV_NETWORK_DEVICE_PLUGIN_REPO"
+    git clone $SRIOV_NETWORK_DEVICE_PLUGIN_REPO $WORKSPACE/sriov-network-device-plugin
+    pushd $WORKSPACE/sriov-network-device-plugin
+    git checkout $SRIOV_NETWORK_DEVICE_PLUGIN_BRANCH
+    git log -p -1 > $ARTIFACTS/sriov-network-device-plugin.txt
+    make build
+    popd
+    mkdir -p /etc/pcidp/
+    cat > /etc/pcidp/config.json <<EOF
+{
+    "resourceList": [{
+        "resourceName": "sriov",
+        "selectors": {
+                "vendors": ["15b3"],
+                "devices": ["1018"],
+                "drivers": ["mlx5_core"]
+            }
+    }
+    ]
+}
+EOF
+    cp /etc/pcidp/config.json $ARTIFACTS
+    return 0
+}
+
+
+function pod_create {
+    cat > /tmp/sriov_pod.yaml <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mofed-test-pod-1
+  annotations:
+    k8s.v1.cni.cncf.io/networks: sriov-net1
+spec:
+  restartPolicy: OnFailure
+  containers:
+    - image: mellanox/mlnx_ofed_linux-4.4-1.0.0.0-centos7.4
+      name: mofed-test-ctr
+      imagePullPolicy: IfNotPresent
+      securityContext:
+        capabilities:
+          add: ["IPC_LOCK"]
```

Comment:
It seems that `test_pod` only runs `ip a` cmd to test the pod, do we need `IPC_LOCK` capability?